misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ServiceMemberFunctionNodeVisitor.java

```diff
@@ -390,4 +396,39 @@ private ClassSymbol getReferredClassSymbol(TypeSymbol symbol) {
         }
         return classSymbol;
     }
+
+    private boolean isAnyReferredActionNode(ObjectFieldNode objectFieldNode) {
+        Optional<Symbol> objFieldNodeSymbol = semanticModel.symbol(objectFieldNode);
+        if (objFieldNodeSymbol.isEmpty()) {
+            return false;
+        }
+        boolean isAnyReferredActionNode = false;
+        List<LineRange> objFieldNodeRefs = semanticModel.references(objFieldNodeSymbol.get())
+                .stream().map(Location::lineRange).collect(Collectors.toList());
+        objFieldNodeRefsLoop:
+        for (LineRange lineRange : objFieldNodeRefs) {
+            Node referredNode = findNode(syntaxTree, lineRange);
+            while (!referredNode.kind().equals(SyntaxKind.SERVICE_DECLARATION) &&
+                    !referredNode.kind().equals(SyntaxKind.MODULE_PART)) {
+                if (referredNode.kind().equals(SyntaxKind.REMOTE_METHOD_CALL_ACTION) ||
+                        referredNode.kind().equals(SyntaxKind.CLIENT_RESOURCE_ACCESS_ACTION)) {
+                    isAnyReferredActionNode = true;
+                    break objFieldNodeRefsLoop;
+                }
+                referredNode = referredNode.parent();
+            }
+        }
+        return isAnyReferredActionNode;
+    }
+
+    private NonTerminalNode findNode(SyntaxTree syntaxTree, LineRange lineRange) {
```

Comment:
Is it possible to move this into the utils class, since IMO this can have more usages as a generic util?

Reply:
Yeah, will do that way 👍